client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala

```diff
@@ -1298,24 +1315,73 @@ class LifecycleManager(val appUniqueId: String, val conf: CelebornConf) extends
       shuffleId: Int,
       slotsToDestroy: WorkerResource): Unit = {
     val shuffleKey = Utils.makeShuffleKey(appUniqueId, shuffleId)
-    val parallelism = Math.min(Math.max(1, slotsToDestroy.size()), conf.clientRpcMaxParallelism)
-    ThreadUtils.parmap(
-      slotsToDestroy.asScala,
-      "DestroySlot",
-      parallelism) { case (workerInfo, (primaryLocations, replicaLocations)) =>
-      val destroy = DestroyWorkerSlots(
-        shuffleKey,
-        primaryLocations.asScala.map(_.getUniqueId).asJava,
-        replicaLocations.asScala.map(_.getUniqueId).asJava)
-      var res = requestWorkerDestroySlots(workerInfo.endpoint, destroy)
-      if (res.status != StatusCode.SUCCESS) {
-        logDebug(s"Request $destroy return ${res.status} for $shuffleKey, " +
-          s"will retry request destroy.")
-        res = requestWorkerDestroySlots(
-          workerInfo.endpoint,
-          DestroyWorkerSlots(shuffleKey, res.failedPrimarys, res.failedReplicas))
+
+    def retryDestroy(status: DestroyFutureWithStatus, currentTime: Long): Unit = {
+      status.retryTimes += 1
+      status.startTime = currentTime
+      status.future =
+        status.endpoint.ask[DestroyWorkerSlotsResponse](status.message)
+    }
+
+    val startTime = System.currentTimeMillis()
+    val futures = new util.LinkedList[DestroyFutureWithStatus]()
+    slotsToDestroy.asScala foreach { case (workerInfo, (primaryLocations, replicaLocations)) =>
```

Comment:
```suggestion
    slotsToDestroy.asScala.foreach { case (workerInfo, (primaryLocations, replicaLocations)) =>
```
https://github.com/databricks/scala-style-guide?tab=readme-ov-file#infix-methods

Reply:
done